libbeat/filter/registry.go

```diff
@@ -0,0 +1,29 @@
+package filter
+
+import (
+	"fmt"
+
+	"github.com/elastic/beats/libbeat/common"
+	"github.com/elastic/beats/libbeat/logp"
+)
+
+type FilterRule interface {
+	Filter(event common.MapStr) (common.MapStr, error)
+	CheckConfig(c common.Config) error
```

Comment:
not really, but I added it there, just that we don't forget about it to implement. Shall I remove it?